### PR TITLE
Fix or remove bogus yaml definitions from the default mods.

### LINKED ISSUE #15232: Death Hand missile launch notification is missing
#15201 reveals that #11483 fudged the notification definition for the Death Hand missile (`MissileLaunchDetected: MissileLaunchDetected`).  This should instead be using `LaunchSpeechNotification`.

### LINKED ISSUE #15233: nod03b defines the wrong tooltip for the comm center
Regression from #12141: the description remained on `Tooltip` where it was ignored until revealed by #15201.

### LINKED ISSUE #15234: HARV.FullHusk is never used
`HarvesterHuskModifier` defines `FullActor`, but it should actually be `FullHuskActor`, so the override is never applied.  Revealed by #15201.

### LINKED ISSUE #15236: RA's BOXES01 actor is listed as a tree in the map editor.
`Categories: Decoration` is defined on the `Tooltip` trait instead of `EditorTilesetFilter`.  Revealed by #15201.

### LINKED ISSUE #15250: The THF is using outdated "InfiltrateTypes" instead of "Types"
RA: The THF (currently disabled unit) uses an outdated trait name; "InfiltrateTypes:" should be changed to "Types:"
https://github.com/OpenRA/OpenRA/blob/0fc466daee53d4e0e3c157ce023844e311644dc5/mods/ra/rules/infantry.yaml#L508-L509
https://github.com/OpenRA/OpenRA/wiki/Traits#infiltrates

### PR DESCRIPTION
**Pull request by pchote**
This PR removes all of the unused/bogus trait and weapon property definitions in the default mods. See commit descriptions for more info where required, and feel free to use my [throw-on-junk-properties](https://github.com/pchote/OpenRA/tree/throw-on-junk-properties) branch to verify that there are no more hidden cases (except for the AI support power decisions, which will have a PR of their own).

Fixes #15232.
Fixes #15233.
Fixes #15234.
Fixes #15236.
Fixes #15250.

### CODE REVIEW

File: mods/d2k/rules/structures.yaml [hunk 1]
```diff
@@ -596,7 +596,6 @@ outpost:
 		RequiresCondition: !severe-damaged
 	GrantConditionOnDamageState@STOPDISH:
 		Condition: severe-damaged
-		ValidDamageState: Medium, Heavy, Critical
```
Comment (pchote) [inline mods/d2k/rules/structures.yaml]: The correct yaml name here would be `ValidDamageStates`, but this then produces the wrong behaviour ingame.  We don't want to stop the dish on `Medium`, only `Heavy` and `Critical`, which is already the default.

File: mods/d2k/rules/structures.yaml [hunk 2]
```diff
@@ -1130,7 +1129,6 @@ palace:
 		RequiresCondition: !launchpad-damaged && harkonnen
 	GrantConditionOnDamageState@LAUNCHPADDAMAGED:
 		Condition: launchpad-damaged
-		ValidDamageState: Medium, Heavy, Critical
```
Comment (pchote) [inline mods/d2k/rules/structures.yaml]: Likewise here.

File: mods/ra/rules/infantry.yaml [hunk 3]
```diff
@@ -506,7 +506,6 @@ THF:
 	Passenger:
 		PipType: Yellow
 	Infiltrates:
-		InfiltrateTypes: Cash
```
Comment (pchote) [inline mods/ra/rules/infantry.yaml]: We don't define Cash as an infiltratable types, so renaming this to `Types` would still have been wrong.

File: mods/ra/rules/infantry.yaml [hunk 4]
```diff
@@ -706,7 +705,6 @@ Ant:
 	Mobile:
 		Speed: 99
 		TurnSpeed: 12
-		SharesCell: no
```
Comment (pchote) [inline mods/ra/rules/infantry.yaml]: Related to #15235, which is not fixed here.  This PR just removes the bogus definition.

File: mods/d2k/rules/defaults.yaml [hunk 5]
```diff
@@ -367,7 +367,6 @@
 	Selectable:
 		Priority: 2
 	RevealsShroud:
-		VisibilityType: CenterPosition
```
Comment (pchote) [inline mods/d2k/rules/defaults.yaml]: Related to #15231, which is not fixed here.  This PR just removes the bogus definition.

**Review by reaperrr — APPROVED** [review]

**Review by Smittytron — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/09c135972c48564a8a97308f3b4729ec52a84e09...749dee54ba2c4d18a184536b191962a2840fd98a

**Pull request merged**